### PR TITLE
Cleanup nix config

### LINKED ISSUE #64: Build failure when building with Nix
Faced with problem when tried to build

```
nix-build --show-trace
copying path '/nix/store/aqr5jws27ira7mmi884r4n72yd2iqc34-happy-1.19.9' from 'https://cache.nixos.org'...
building '/nix/store/q325bf3n7y33nszmsn9pgcnxbp3dpava-python3-3.7.3.drv'...
unpacking sources
unpacking source archive /nix/store/r5ik1sbafp0kr3a7prnaca1lmmqmb0b6-Python-3.7.3.tar.xz
source root is Python-3.7.3
setting SOURCE_DATE_EPOCH to timestamp 1553547549 of file Python-3.7.3/Misc/NEWS
patching sources
applying patch /nix/store/gd8ycd6j3qhcq1qccaqz68zdq5l4c2wf-no-ldconfig.patch
patching file Lib/ctypes/util.py
applying patch /nix/store/ggk7z3dvydrkypxff7d7c6v1slvbriin-darwin-libutil.patch
patching file Modules/posixmodule.c
Hunk #1 succeeded at 5907 (offset 27 lines).
applying patch /nix/store/9kwzs3pplms8sijf55sdryypzvic4x1s-python-3.x-distutils-C++.patch
patching file Lib/_osx_support.py
patching file Lib/distutils/cygwinccompiler.py
patching file Lib/distutils/sysconfig.py
Hunk #1 succeeded at 183 (offset 13 lines).
Hunk #2 succeeded at 202 (offset 13 lines).
Hunk #3 succeeded at 231 (offset 13 lines).
patching file Lib/distutils/unixccompiler.py
Hunk #3 FAILED at 183.
1 out of 3 hunks FAILED -- saving rejects to file Lib/distutils/unixccompiler.py.rej
patching file Makefile.pre.in
Hunk #1 succeeded at 615 (offset 31 lines).
builder for '/nix/store/q325bf3n7y33nszmsn9pgcnxbp3dpava-python3-3.7.3.drv' failed with exit code 1
cannot build derivation '/nix/store/26rq8gd6dh6vzn9rzz56ljqdr1sq90wi-ghc-8.6.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/6sxn76pka3xr30c1wg49q01qpkzy9jdh-cabal2nix-2.14.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/49l07msc317w2gijh3mns7jr2wamaxii-cabal2nix-ormolu.drv': 1 dependencies couldn't be built
error: while evaluating the attribute 'ormolu' at ~/tmp/ormolu/default.nix:14:13:
while evaluating 'callCabal2nix' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:206:32, called from /Users/pachin/tmp/ormolu/default.nix:14:24:
while evaluating 'callCabal2nixWithOptions' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:191:66, called from /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:206:38:
while evaluating 'overrideCabal' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/lib.nix:37:24, called from /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:202:10:
while evaluating 'callPackageKeepDeriver' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:156:33, called from /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:202:25:
while evaluating 'overrideCabal' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/lib.nix:37:24, called from /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:157:5:
while evaluating 'callPackage' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:114:22, called from /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:157:20:
while evaluating 'callPackageWithScope' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:74:37, called from /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:114:28:
while evaluating 'makeOverridable' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/lib/customisation.nix:67:24, called from /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:97:8:
while evaluating 'drvScope' at /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/pkgs/development/haskell-modules/make-package-set.nix:87:18, called from /nix/store/72vjs9dvrj8nzsavl0i6j3n6dv62i8jl-source/lib/customisation.nix:69:12:
build of '/nix/store/49l07msc317w2gijh3mns7jr2wamaxii-cabal2nix-ormolu.drv' failed
```

### PR DESCRIPTION
**Pull request by luc-tielen**
Fixes #64 (tested on my macbook).

- Fixed it by pointing to a newer nixpkgs commit (the commit you pointed to previously is broken on darwin).
- Restructured default.nix and added a shell.nix (now people can use `lorri watch` too if they wish to do so).
- Ran `nixfmt` to give it consistent formatting

All nix commands should stay the same.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,24 +1,20 @@
...
-       { packages = (ps: [ ps.ormolu ]);
-         buildInputs = [
-           pkgs.cabal-install
-         ];
-       }
-     else haskellPackages.ormolu
+let
+  pkgs = import ./nix/nixpkgs;
+  compiler = "ghc864";
+  sourceRegex =
```
[the remark this answers is not in the document]
Comment (luc-tielen) [reply default.nix]: Hi @domenkozar, thanks for the tip 😃.
I haven't used nix for that long yet, so feedback welcome 🙂.

[the remark this answers is not in the document]
Comment (domenkozar) [reply default.nix]: Note that you're using wrong nix-gitignore, while it should be https://github.com/hercules-ci/gitignore

**Comment by luc-tielen** [comment]
I noticed that `lorri watch` is now broken for some reason with latest commit (would be nice if that kept working). I will look into that later today.

File: nix/gitignore/default.nix [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{ lib }:
+
+let
+  rev = "ec5dd0536a5e4c3a99c797b86180f7261197c124";
```
Comment (mrkkrp) [inline nix/gitignore/default.nix]: Maybe this stuff could be added to `nixpkgs` itself?

**Comment by mrkkrp** [comment]
Is this ready or not?

**Comment by mrkkrp** [comment]
I merged for now. Open another one if you want to adjust something.

**Comment by luc-tielen** [comment]
Thanks for merging it in! 

Problem with lorri seems to be related Mac's fsevents API, not with this repo. So I think it's fine for now.

**Comment by mrkkrp** [comment]
The `.gitignore` based source filter causes issues because it includes too much, so changes to unrelated files cause re-compilation of Ormolu. I'm going to replace it with a simple regexp as it was before.

**Comment by domenkozar** [comment]
You should be able yo compose the filter to exclude extra files i.e. data

**Comment by mrkkrp** [comment]
There is no overlap between the files that regexp selects and the files that are excluded by `.gitignore` anyway.

**Pull request merged**